### ISSUE TITLE
ClassNotFoundException while calling AWS S3 Upload

### ISSUE THREAD

**Issue by romanleicht** [DataDog/dd-trace-java#244]
When I upload to S3 this happens:
```
Caused by: java.lang.NoClassDefFoundError: datadog/trace/agent/deps/fasterxml/jackson/databind/ObjectMapper
	at datadog.trace.instrumentation.aws.SpanDecorator.<clinit>(SpanDecorator.java:29)
	at datadog.trace.instrumentation.aws.TracingRequestHandler.beforeRequest(TracingRequestHandler.java:74)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:770)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:724)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:717)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:699)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:667)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:649)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:513)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4325)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4272)
	at com.amazonaws.services.s3.AmazonS3Client.putObject(AmazonS3Client.java:1749)
	at com.amazonaws.services.s3.AmazonS3Client.putObject(AmazonS3Client.java:1611)

Caused by: java.lang.ClassNotFoundException: datadog.trace.agent.deps.fasterxml.jackson.databind.ObjectMapper
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381) ~[na:1.8.0_162]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[na:1.8.0_162]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:338) ~[na:1.8.0_162]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_162]
```

### Environment
OracleJDK 1.8.0_162
com.amazonaws:aws-java-sdk-s3:1.11.263
com.datadoghq:dd-java-agent:0.4.0

**Comment by tylerbenson** [comment]
Thanks for the report.  We'll take a look.

**Comment by dashbar** [comment]
Having a same issue with DynamoDB. I think ClassLoader is not functioning very well.

**AWS SDK Version**
```
        <dependency>
            <groupId>com.amazonaws</groupId>
            <artifactId>aws-java-sdk-dynamodb</artifactId>
            <version>1.11.264</version>
        </dependency>
```

**Java Agent Version**
```
dd-java-agent - version: 0.4.0~0bfa65f
```

**JVM Version**
```
root@07882d0d878e:~/bo# java -version
openjdk version "1.8.0_151"
OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-1~deb9u1-b12)
OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
```

**Running Command**
```
java -Xms1M -Xmx1G -javaagent:dd-java-agent.jar -jar OUR_JAR
```

**Exception**
```
service-accounts_1     | Exception in thread "Thread-0" java.lang.NoClassDefFoundError: datadog/trace/agent/deps/fasterxml/jackson/core/JsonProcessingException
service-accounts_1     | 	at datadog.trace.instrumentation.aws.TracingRequestHandler.beforeRequest(TracingRequestHandler.java:74)
service-accounts_1     | 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:770)
service-accounts_1     | 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:724)
service-accounts_1     | 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:717)
service-accounts_1     | 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:699)
service-accounts_1     | 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:667)
service-accounts_1     | 	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:649)
service-accounts_1     | 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:513)
service-accounts_1     | 	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.doInvoke(AmazonDynamoDBClient.java:2925)
service-accounts_1     | 	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.invoke(AmazonDynamoDBClient.java:2901)
service-accounts_1     | 	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.executeListTables(AmazonDynamoDBClient.java:1812)
service-accounts_1     | 	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.listTables(AmazonDynamoDBClient.java:1788)
service-accounts_1     | 	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.listTables(AmazonDynamoDBClient.java:1824)
```

**Comment by zzFluke** [comment]
same issue

**Comment by realark** [comment]
@romanleicht @dashbar @zzFluke Thanks for reporting this. Can any of you try a fix on a dev machine? You can drop the snapshot jar in over your existing `dd-java-agent.jar`

https://2246-89221572-gh.circle-artifacts.com/0/root/dd-trace-java/build/dd-java-agent/build/libs/dd-java-agent-0.4.1-SNAPSHOT.jar

**Comment by romanleicht** [comment]
works on my machine :)

**Comment by realark** [comment]
Thanks for verifying @romanleicht. Fix released in `0.4.1` of dd-trace-java. Upgrading to the latest release will resolve this issue.
